src/core/net/ip6.hpp

```diff
@@ -196,7 +196,7 @@ class Ip6 : public InstanceLocator, private NonCopyable
      * processing is complete, including when a value other than `kErrorNone` is returned.
      *
      * @param[in]  aMessage          A reference to the message.
-     * @param[in]  aFromNcpHost      If the message was forwarded from the NCP Host.
+     * @param[in]  aFromHost         If the message was forwarded from the host.
```

Comment:
```suggestion
     * @param[in]  aForwardedFromHost         If the message was forwarded from the host.
```

Reply:
The term `Forward` makes sense if we have another netif and the message is forwarded from it to the OT stack. But OT stack may be used on SoC type use-case where there is no other network interface other than OT Thread netif itself and then we can use `aFromHost` to indicate the message is generated/originated by the host application and passed down to OT stack to be sent. I think in such a case, the term "forward" may not fit well. I updated the documentation for the param to make it clear (in the new push).

Reply:
> The term `Forward` makes sense if we have another netif and the message is forwarded from it to the OT stack.

There is another netif specified by Thread Spec, which is the Backbone link or adjacent infrastructure link. What I mean is that there is implicitly a Backbone link defined even if there is no `netif` structure defined in OT code for that. 

> But OT stack may be used on SoC type use-case where there is no other network interface other than OT Thread netif itself and then we can use aFromHost to indicate the message is generated/originated by the host application and passed down to OT stack to be sent.

If we check the usage of `aFromNcpHost` throughout the OT code, there is only one usage besides `DUA` code:
- In `ProcessReceiveCallback`, we reject to forward the message back to Host if `aFromNcpHost` is true

However, we already know that a BR **SHOULD** forward the message back to Host if the message was initiated by the Host itself, no matter it's RCP or NCP architecture. 

I think we introduced the `aFromNcpHost` to tell if the message is passed from Host, however that's implementation-specific and not what we actually need. We need to tell if the message is forwarded from another netif, in which case we should never forward the message back to Host.

Reply:
We intended the term "host" in `aFromHost` to represent the entity that is controlling OT stack (and calling its API). It's basically user of OT APIs, e.g. `otbr` or `wpantund` for routers/BR or in case of simple SED application, the sw entity controlling OT APIs). We can consider different name, say `aFromUser`? (other name suggestion are welcome. I myself prefer `aFromHost` a bit more over `aFromUser`).

The original intention behind the `aFromHost` was to indicate if a message is originated from user/host or if it was received by the Thread radio (originated from Thread stack). Its main purpose was to filter out and not pass back a messages submitted by user back to the user in the callback `ProcessReceiveCallback`.
- This was equally applicable (and useful) for router/BR style devices and for SED type devices.
   - Particularly, on a simple SED application, we may not want to pass back a user submitted IPv6 message back to the user in `ProcessReceiveCallback()` callback since the callback implementation on a such device may be simple and assumes that the message is received over radio.
     
- This was all added in Thread 1.1 and before the notions of backbone link, infra structure link, etc.
- I think we figured out with some new 1.2/1.3 use-case, we may want to relax such filtering and allow messages from user/host to be passed back to it in certain situations (changes from #7187).
- I think we need to have a more elaborate model (basically a single boolean is not longer sufficient for all situations). Probably need a `enum` to inidcate the oirgin of IPv6 message. This is something that needs more thoughts and considerations. I would consider this and other enhancements for future PRs.
- The goal of this PR is only to do a simple rename (to make the name more generic and clear).

Reply:
> Particularly, on a simple SED application, we may not want to pass back a user submitted IPv6 message back to the user in ProcessReceiveCallback() callback since the callback implementation on a such device may be simple and assumes that the message is received over radio.

I think SED may still pass the message back to Host if the IPv6 destination is itself.

> I think we need to have a more elaborate model (basically a single boolean is no longer sufficient for all situations). Probably need a enum to inidcate the oirgin of IPv6 message. This is something that needs more thoughts and considerations. We should consider this and other enhancements in future PRs.

Agree with that. 

---

I think that what we really need in actual code seems to be checking whether the message was forwarded from another interface. And we rarely want to change OT behavior according to whether the message was passed from Host, which is an architecture-specific information. 

That being said, rename to `aFromHost` for now seems good to me.